### PR TITLE
Help mailers

### PR DESCRIPTION
**Pull request by liannivin**

### CODE REVIEW

File: app/mailers/rating_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class RatingMailer < ApplicationMailer
+    default from: 'notifications@myblog.com'
+  
+    def report_most_popular_songs
+      @user = params[:user]
+      @songs = Song.joins(:ratings).group('songs.id').order('COUNT(ratings.id) DESC').limit(3)
```
Comment (condef5) [inline app/mailers/rating_mailer.rb:6]: You should use the `Arel` in your queries for security reasons.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  after_create :send_registration_mail, :welcome
+  has_many :providers
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: %i[facebook github]
+
+  def role?(expected_role)
```
Comment (condef5) [inline app/models/user.rb:8]: I like this method, very good example of how to use `dry`.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,38 @@
...
+    user
+  end
+
+  def send_registration_mail
+    User.where(role: "admin").each do |user|
+      UserMailer.with(user: user) .user_created.deliver_later
+    end
+  end
+
+  def regular?
```
Comment (condef5) [inline app/models/user.rb:27]: Why you use `regular?` and `admin?` methods?, if you already have the method `role?`

File: config/environments/development.rb [hunk 4]
```diff
@@ -52,6 +52,21 @@
...
 
+  # ActionMailer config
+config.action_mailer.delivery_method = :smtp
+config.action_mailer.smtp_settings =
+  {
+    address: 'smtp.gmail.com',
+    port: 587,
+    domain: 'gmail.com',
+    user_name: 'liamrn94',
+    password: 'yqgticrjytcwrevk',
```
Comment (condef5) [inline config/environments/development.rb:63]: You could use the enviroment variables for `user_name` and `password` :).

File: app/models/user.rb (hunk 2 above)
Comment (Linzeur) [reply app/models/user.rb:8]: A generic method to evaluate different roles, good idea!

**Pull request merged**